contracts/clients-tests/BlsProvider.test.ts

```diff
@@ -0,0 +1,535 @@
+/* eslint-disable camelcase */
+import { ethers as hardhatEthers } from "hardhat";
+import chai, { expect } from "chai";
+import spies from "chai-spies";
+import { ethers, Wallet } from "ethers";
+import { parseEther, formatEther, id } from "ethers/lib/utils";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+import {
+  BlsProvider,
+  BlsSigner,
+  BlsWalletWrapper,
+  MockERC20__factory,
+  NetworkConfig,
+} from "../clients/src";
+import getNetworkConfig from "../shared/helpers/getNetworkConfig";
+
+chai.use(spies);
+
+let networkConfig: NetworkConfig;
+let signers: SignerWithAddress[];
+
+let aggregatorUrl: string;
+let verificationGateway: string;
+let rpcUrl: string;
+let network: ethers.providers.Networkish;
+
+let privateKey: string;
+let blsProvider: BlsProvider;
+let blsSigner: BlsSigner;
+
+let regularProvider: ethers.providers.JsonRpcProvider;
+let regularSigner: ethers.providers.JsonRpcSigner;
+
+describe("BlsProvider", () => {
+  beforeEach(async () => {
+    networkConfig = await getNetworkConfig("local");
+    signers = await hardhatEthers.getSigners();
+
+    aggregatorUrl = "http://localhost:3000";
+    verificationGateway = networkConfig.addresses.verificationGateway;
+    rpcUrl = "http://localhost:8545";
+    network = {
+      name: "localhost",
+      chainId: 0x7a69,
+    };
+    privateKey = Wallet.createRandom().privateKey;
```

Comment:
It is generally not a good idea to use purely random values in tests as it can make it hard to reproduce failures with the random value. I would just hard code a random private key here. Could also just use one of the well known Hardhat ones.

Reply:
Gotcha, changed